packages/babel-helpers/src/helpers.js

```diff
@@ -447,17 +447,27 @@ helpers.inheritsLoose = () => template.program.ast`
 helpers.wrapNativeSuper = () => template.program.ast`
   var _gPO = Object.getPrototypeOf || function _gPO(o) { return o.__proto__ };
   var _sPO = Object.setPrototypeOf || function _sPO(o, p) { o.__proto__ = p; return o };
-  var _construct = (typeof Reflect === "object" && Reflect.construct) ||
-    function _construct(Parent, args, Class) {
-      var Constructor, a = [null];
-      a.push.apply(a, args);
-      Constructor = Parent.bind.apply(Parent, a);
-      return _sPO(new Constructor, Class.prototype);
-    };
+
+  var _construct = typeof Reflect === "object" && Reflect.construct
+    ? function _construct(Parent, args, Class) {
+        // This wrapper is needed because Reflect.construct can't be properly
+        // polyfilled, thus core-js doesn't set the correct __proto__.
+        var result = Reflect.construct(Parent, args, Class);
+        if (!(result instanceof Class)) result = _sPO(result, Class.prototype);
```

Comment:
> (By @jridgewell )
>
> This seems like a core-js bug, not ours.

It's something that `core-js` can't safely polyfill, but we can because we know that the super constructor is a built-in constructor, so its return value hasn't been overridden.

e.g.
```
function A() {
  return {};
}

function Wrapper() {
  return _construct(A, [], Wrapper);
}
Wrapper.prototype = Object.create(...);

// This should be false, but if core-js always did setPrototypeOf it would be true.
new Wrapper() instanceof Wrapper;
```